### PR TITLE
Introduce GitHub Actions PR Check job, fixes JBIDE-27702

### PR DESCRIPTION
**Pull request by odockal**

### CODE REVIEW

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        key: jbosstools-${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          jbosstools-${{ runner.os }}-maven-
+
+    # Build and compile using Maven
+    - name: Build/Compile and run unit tests
+      uses: GabrielBB/xvfb-action@v1
+      with:
+        run: >
+          mvn clean verify -U -fae --settings build-ci/maven-settings.xml -DskipITests=true
```
Comment (sbouchet) [inline .github/workflows/build_test.yml:52]: why the --no-transfer-progress param is not used here ?

Comment (odockal) [reply .github/workflows/build_test.yml:52]: So I see that I did not use --no-transfer-progress when I used --quiet... quite is a bit below this line

**Comment by odockal** [comment]
@sbouchet review please.

File: .github/workflows/build_test.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      matrix:
+        os: [ubuntu-latest]
+        java: ["11"]
+      fail-fast: false
+
+    steps:
+    - name: Checkout PR branch
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
```
Comment (sbouchet) [inline .github/workflows/build_test.yml:21]: please add fetch-depth: 0 to avoid baseline errors

File: .github/workflows/build_test.yml (hunk 1 above)
Comment (sbouchet) [reply .github/workflows/build_test.yml:52]: replace the --quiet by --no-transfer-progress to better read logs.

**Review by sbouchet — CHANGES_REQUESTED** [review]

File: .github/workflows/build_test.yml (hunk 1 above)
Comment (odockal) [reply .github/workflows/build_test.yml:53]: done

**Review by sbouchet — APPROVED** [review]
LGTM.

**Pull request merged**